addons/web_editor/controllers/main.py

```diff
@@ -507,17 +510,20 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
     @http.route(['/web_editor/shape/<module>/<path:filename>'], type='http', auth="public", website=True)
     def background_shape(self, module, filename, **kwargs):
         """
-        Returns a color-customized background-shape.
+        Returns a color-customized svg (background shape or illustration).
         """
-        shape_path = get_resource_path(module, 'static', 'shapes', filename)
-        if shape_path:
-            svg = open(shape_path, 'r').read()
-        else:
-            # Fallback to attachments for future migrations
-            attachment = request.env['ir.attachment'].search([('url', '=like', request.httprequest.path)], limit=1)
+        svg = None
+        if module == 'illustration':
+            attachment = request.env['ir.attachment'].sudo().search([('url', '=like', request.httprequest.path), ('public', '=', True)], limit=1)
```

Comment:
Public user cannot search attachments without sudo, even with public = True in the domain. Is this not the expected behavior?

Reply:
yes it is the case since this commit: 62c9dedafda0cbdc618641eaa80c49535ab0b5f6

Seems legit with the is_public = True, but not sure how your previous code was working in public user in this case.